api/V1/Group.js

```diff
@@ -69,4 +71,154 @@ module.exports = function(app, baseUrl) {
         }
       });
   });
+
+  /**
+   * @api {get} /api/v1/groups Get groups
+   * @apiName GetGroups
+   * @apiGroup Group
+   *
+   * @apiParam {JSON} where [Sequelize where conditions](http://docs.sequelizejs.com/manual/tutorial/querying.html#where) for query.
+   * @apiParam {Number} [userId] Only get groups that this user belongs to.
+   *
+   * @apiUse V1ResponseSuccess
+   *
+   * @apiUse V1ResponseError
+   */
+  app.get(
+    apiUrl,
+    async function(request, response) {
+      log.info(request.method + ' Request: ' + request.url);
+
+      var where = request.query.where;
+      var queryUserId = request.query.userId;
+      try {
+        where = JSON.parse(where);
+      } catch (e) {
+        return responseUtil.send(response, {
+          statusCode: 400,
+          success: false,
+          messages: ['Could not parse "where" to a JSON'],
+        });
+      }
+
+      var userWhere = null;
+      if (queryUserId != null) {
+        userWhere = { id: queryUserId };
+      }
+
+      var groups = await models.Group.findAll({
+        where: where,
+        include: [
+          {
+            model: models.User,
+            attributes: ['id', 'username'],
+            where: userWhere,
+          },
+        ],
+      });
```

Comment:
Up to you whether you change it now or later, but I really do think that all DB queries should be kept in the `models` package so that API handler methods are very thin. They should just parse parameters, call to a model method and then structure the return value.